### ISSUE TITLE
 F8648P V2.1.10P5N15

### ISSUE THREAD

**Issue by lightstal** [alez-repos/ztef8648p#4]
This version of the router will cause the tool to be stuck on [Samba_pwn]: Opening shell and nothing will happen.

![Capture](https://github.com/alez-repos/ztef8648p/assets/50443758/d0e11bd3-fe69-46d5-b963-67ba69d39e00)

**Comment by alez-repos** [comment]
For the moment that router firmware version is not compatible with the tool.

**Comment by codex-20** [comment]
Hi there.

My router has the V2.0.12P7N15 firmware and is also stuck on the Opening shell line.

Do you have any new feedback available since you commented on Feb 12th?

Many thanks for your time and dedication.

**Comment by alez-repos** [comment]
This router version has updated firmware with patched samba server. Sadly I dont have it to test as my router is freezed in previous version.

I could only continue investigating if someone donates one of these routers with that firmware version, or network hardware so I can replace and upgrade mine. Still, there are no guarantees that I'll be able to accomplish anything with that version.

**Comment by codex-20** [comment]
Thank you for your latest message. That's unfortunate! Hopefully you will be able to work on it the near future.

Thanks again!

**Comment by gorbypark** [comment]
I also just got a router installed and it is V2.0.12P7N15.

> This router version has updated firmware with patched samba server. Sadly I don't have it to test as my router is freezed in previous version.

I don't know much python, but looking at samba_pwn it seems it was just logging in as "test"?
```python
smbClient.login("test", "test", "", "", "")
```
was the the "hole" that was patched, they just had some test credentials enabled?

**Comment by alez-repos** [comment]
No, if you follow the flow of the code you will see that this user is created (via http post) prior to attemp a login. I choose test/test just randomly, any other name would have worked.

The flaw that allowed my program to work was a lack of management of symbolic links at the router samba server, allowing to extend the configuration. This has been patched in new firmware version like the one you have.

**Comment by gorbypark** [comment]
> No, if you follow the flow of the code you will see that this user is created (via http post) prior to attemp a login. I choose test/test just randomly, any other name would have worked.
> 
> The flaw that allowed my program to work was a lack of management of symbolic links at the router samba server, allowing to extend the configuration. This has been patched in new firmware version like the one you have.

aha, it seemed too easy

**Comment by maximpedraza0** [comment]
Buenas Alez estoy en las mismas y el tema de hackeos no controlo nada... ¿Te interesaría q te montase una RPI con ssh accesible para q puedas mirar y hacer pruebas?

**Comment by Ardi-mounsif** [comment]
Does anyone know how to downgrade firmware?

**Comment by Damrod** [comment]
> This router version has updated firmware with patched samba server. Sadly I dont have it to test as my router is freezed in previous version.
> 
> I could only continue investigating if someone donates one of these routers with that firmware version, or network hardware so I can replace and upgrade mine. Still, there are no guarantees that I'll be able to accomplish anything with that version.

I'm willing to collaborate on this topic. My router is the new version. I've tried to use the same vulnerability with the ftp server (I won't have the preexec hook, but at least I could have managed to browse the root of the router's filesystem, and maybe that would have been enough, because then I would have been able to put the samba config there with the preexec hook), but i will see it listed as a symlink to 'unsupported reparse tag 0xa000000c'. If I try to create the symlink there, it will error with "/mnt/digirouter/usb1_1_1/raiz" -> "/" failed: Function not implemented (38). I've also tried several other filesystems that support symlinks (extFAT, UDF, ext4) in the router, and I've been unable to have the file servers acknowledge them.

The router has yet another file server: DMS/DLNA. But I have no idea how does that work and haven't really tried.

I'm kind of stuck.

The only thing I can come up with, besides the DMS/DLNA server, is try to open it up and connect a serial cable to it. But I won't do it: I don't technically OWN this router.
